zendriver/core/expect.py

```diff
@@ -0,0 +1,185 @@
+
+    @property
+    async def response_body(self):
+        """
+        Get the body of the matched response.
+        :return: The response body.
+        :rtype: str
+        """
+        request_id = (await self.request_future).request_id
+        body = await self.tab.send(cdp.network.get_response_body(request_id=request_id))
+        return body
+
+
+class RequestExpectation(BaseRequestExpectation):
+    """
+    Class for handling request expectations.
+    This class extends `BaseRequestExpectation` and provides a property to access the matched request.
+    :param tab: The Tab instance to monitor.
+    :type tab: Tab
+    :param url_pattern: The URL pattern to match requests.
+    :type url_pattern: Union[str, re.Pattern[str]]
+    """
+
+    @property
+    async def value(self) -> cdp.network.RequestWillBeSent:
+        """
+        Get the matched request event.
+        :return: The matched request event.
+        :rtype: cdp.network.RequestWillBeSent
+        """
+        return await self.request_future
+
+
+class ResponseExpectation(BaseRequestExpectation):
+    """
+    Class for handling response expectations.
+    This class extends `BaseRequestExpectation` and provides a property to access the matched response.
+    :param tab: The Tab instance to monitor.
+    :type tab: Tab
+    :param url_pattern: The URL pattern to match responses.
+    :type url_pattern: Union[str, re.Pattern[str]]
+    """
+
+    @property
+    async def value(self) -> cdp.network.ResponseReceived:
+        """
+        Get the matched response event.
+        :return: The matched response event.
+        :rtype: cdp.network.ResponseReceived
+        """
+        return await self.response_future
+
+
+class DownloadExpectation:
+    def __init__(self, tab: Connection):
+        self.tab = tab
+        self.future: asyncio.Future[cdp.page.DownloadWillBegin] = asyncio.Future()
+        # TODO: Improve
+        self.default_behavior = (
+            self.tab._download_behavior[0] if self.tab._download_behavior else "default"
+        )
+
+    async def _handler(self, event: cdp.page.DownloadWillBegin):
+        self._remove_handler()
+        self.future.set_result(event)
+
+    def _remove_handler(self):
+        self.tab.remove_handlers(cdp.page.DownloadWillBegin, self._handler)
+
+    async def __aenter__(self):
+        """
+        Enter the context manager, adding download handler, set download behavior to deny.
+        """
+        await self.tab.send(cdp.browser.set_download_behavior(behavior="deny"))
+        self.tab.add_handler(cdp.page.DownloadWillBegin, self._handler)
+        return self
+
+    async def __aexit__(self, *args):
+        """
+        Exit the context manager, removing handler, set download behavior to default.
+        """
+        await self.tab.send(
+            cdp.browser.set_download_behavior(behavior=self.default_behavior)
+        )
+        self._remove_handler()
+
+    @property
+    async def value(self):
```

Comment:
just noticed this as well
```suggestion
    async def value(self): -> cdp.page.DownloadWillBegin
```